examples/with-react-ga/components/Layout.js

```diff
@@ -1,6 +1,11 @@
 import React from 'react'
+import Router from 'next/router'
 import { initGA, logPageView } from '../utils/analytics'
 
+Router.onRouteChangeComplete = (url) => {
+  logPageView()
+}
+
```

Comment:
@fabioespinosa
In next.js routing to new URL will remount this layout component (unless you are using shallow routing AFAIK) so the new pageview will be registered.

By logging pageView both in Router completion event and componentDidMount(of layout component) we may log two page views for client side routing as layout component is remounted every time the route changes. 

This is why I just used componentDidMount in this example to log pageViews which works for most of the time. For [shallow routing](https://github.com/zeit/next.js#shallow-routing) cases you may watch for route change in ` componentWillReceiveProps` and trigger pageview here.

cc @timneutkens